File: survival-graphics/src/main/java/net/survival/graphics/particle/ClientParticleSpace.java
package net.survival.graphics.particle;

import java.util.HashMap;
import java.util.Map;

public class ClientParticleSpace {
    private final HashMap<String, ParticleSim> particleDomains = new HashMap<>();

    public void burstParticles(
            double x,
            double y,
            double z,
            double strength,
            int quantity,
            String texture)
    {
        particleDomains.putIfAbsent(texture, new ParticleSim());
        particleDomains.get(texture).burstParticles(x, y, z, strength, quantity);
    }

    public Map<String, ParticleSim> getParticleDomains() {
        return particleDomains;
    }

    public void step(double elapsedTime) {
        for (var particleDomain : particleDomains.values()) {
            particleDomain.step(elapsedTime);
        }
    }
}

File: survival-graphics/src/main/java/net/survival/graphics/particle/ClientParticleSpace.kt
package net.survival.graphics.particle

import java.util.*

class ClientParticleSpace {
    private val particleSims = HashMap<String, ParticleSim>()

    fun burstParticles(
        x: Double,
        y: Double,
        z: Double,
        strength: Double,
        quantity: Int,
        texture: String
    ) {
        particleSims.putIfAbsent(texture, ParticleSim())
        particleSims[texture]!!.burstParticles(x, y, z, strength, quantity)
    }

    fun getParticleDomains(): Map<String, ParticleSim> {
        return particleSims
    }

    fun step(elapsedTime: Double) {
        for (particleDomain in particleSims.values) {
            particleDomain.step(elapsedTime)
        }
    }
}